gusto/time_discretisation.py

```diff
@@ -471,39 +411,23 @@ def rhs(self):
 
         return r.form
 
-    def solve_stage(self, x_in, stage):
-        """
-        Perform a single stage of the Runge-Kutta scheme.
-
-        Args:
-            x_in (:class:`Function`): field at the start of the stage.
-            stage (int): index of the stage.
-        """
-        if stage == 0:
-            self.solver.solve()
-            self.k1.assign(self.x_out)
-            self.x1.assign(x_in + 0.5 * self.dt * self.k1)
+    def solve_stage(self, x0, stage):
+        self.x1.assign(x0)
+        for i in range(stage):
+            self.x1.assign(self.x1 + self.dt*self.butcher_matrix[stage-1, i]*self.k[i])
+        if self.limiter is not None:
+            self.limiter.apply(self.x1)
+        self.solver.solve()
+        self.k[stage].assign(self.x_out)
 
-        elif stage == 1:
-            for evaluate in self.evaluate_source:
-                evaluate(self.x1, self.dt)
-            self.solver.solve()
-            self.k2.assign(self.x_out)
-            self.x1.assign(x_in + 0.5 * self.dt * self.k2)
+        if (stage == self.nStages - 1):
+            self.x1.assign(x0)
+            for i in range(self.nStages):
```

Comment:
I guess we need to add the physics source evaluation here too?

Reply:
Not required here, as this is the final timestep evaluation of field (after solve)